Restore Flutter gallery license page scrolling

HansMuller opened the pull request:
LazyBlock wasn't updating its scrollable's gesture detector when the length of its contents changed.

A regression test isn't included because the flutter license registry isn't populated for tests.  I've started a prerelease checklist for things that will need to be verified manually.

Fixes #5920

eseidelGoogle commented:
Can/should we just populate the license registry with fake data?  It's pretty common for testing systems to need to include "fixtures" like this.

dragostis reviewed (commented):
Maybe we should add a short test for this?

eseidelGoogle commented:
@yjbanov has also prompted me to offer his services for expanding our testing capabilities if necessary. :)  There is passion within our team to improve our testing status quo.  (I suspect the yegor-ninja is not necessary here, but attempting to get used to my new prompting directives.) :)

Hixie commented:
We should have two sets of tests here.

The first set should be testing a unit test of the actual underlying bug, in an artificial environment.

The second should be a flutter_tools test to verify that the asset for the licenses is being included. That should have been tested a while ago but we should do it here while we're thinking of it.

HansMuller commented:
To write a test, a fix is needed for #5950

Commit:
Restore Flutter gallery license page scrolling

HansMuller commented:
@Hixie I've fixed #5950 and added a test to this PR that checks the LazyBlock bug that caused the problem.

Adding a flutter_tools test that verifies that the asset for the licenses is being included seems like a worthwhile change, but maybe a separate PR? Either that or some advice about how to write such a test.

HansMuller commented:
@eseidel Attempting to get the (presumably empty) licenses registry from a widget test hangs, it's waiting on rootBundle.loadString('LICENSE', cache: false). Evicting LICENSE from the rootBundle doesn't help.

Hixie commented:
![LGTM](http://junkyard.damowmow.com/563)
